File: src/main/java/net/hectus/neobb/turn/default_game/item/TIronShovel.java
package net.hectus.neobb.turn.default_game.item;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterbuffFunction;
import net.hectus.neobb.turn.default_game.block.TCauldron;
import net.hectus.neobb.turn.here_game.HereTurn;
import org.bukkit.Location;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public class TIronShovel extends ItemTurn implements CounterbuffFunction, NeutralClazz {
    public TIronShovel(NeoPlayer player) { super(player); }
    public TIronShovel(ItemStack data, Location location, NeoPlayer player) { super(data, location, player); }

    @Override
    public boolean requiresUsageGuide() { return true; }

    @Override
    public int cost() {
        return 5;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn());
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(turn -> turn instanceof HereTurn, "dirt"), CounterFilter.of(turn -> turn instanceof HereTurn, "grass"), CounterFilter.of(turn -> turn instanceof TCauldron, "cauldron"), CounterFilter.clazz(HotClazz.class)); // TODO: Switch to actual turn classes.
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/item/TIronShovel.kt
package net.hectus.neobb.turn.default_game.item

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.CounterFilter
import net.hectus.neobb.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.turn.default_game.attribute.function.CounterbuffFunction
import net.hectus.neobb.turn.default_game.block.TCauldron
import net.hectus.neobb.turn.default_game.flower.TDirt
import org.bukkit.inventory.ItemStack

class TIronShovel(data: ItemStack?, cord: Cord?, player: NeoPlayer?) : ItemTurn(data, cord, player), CounterbuffFunction, NeutralClazz {
    override val cost: Int = 5

    override fun counters(): List<CounterFilter> {
        return listOf(CounterFilter.of("dirt") { it is TDirt }, CounterFilter.of("cauldron") { it is TCauldron }, CounterFilter.clazz(HotClazz::class))
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(ExtraTurn())
    }
}
